target/linux/ramips/image/mt7621.mk

```diff
@@ -1685,6 +1685,31 @@ define Device/zio_freezio
 endef
 TARGET_DEVICES += zio_freezio
 
+define Device/zyxel_keenetic-giga-iii
+  $(Device/dsa-migration)
+  KERNEL_SIZE := 4096k  # kernel partition size
+  IMAGE_SIZE := 50m
```

Comment:
Please specify in "k" instead, thats common practice.

Reply:
Megabytes are also used:
```
$ grep -r --include "*.mk" -E "IMAGE_SIZE := [0-9]{1,3}m"
layerscape/image/armv8_64b.mk:  IMAGE_SIZE := 64m
layerscape/image/armv7.mk:  IMAGE_SIZE := 64m
ath79/image/nand.mk:  IMAGE_SIZE := 116m
```
And this suffix is handled correctly: https://github.com/openwrt/openwrt/blob/ac03e246351fbfa98e98f0319cce3a8663f836d3/include/image-commands.mk#L171

Reply:
Out of multiple hundred uses of this variable, there is indeed a one-digit number of cases, on other targets, that does not use "k". That's essentially what common practice means.
So, please change it after you have proven my point now.

Reply:
OK. So I'll write the value 51000k

Reply:
You have no clue what this value does, do you?

Reply:
Do you want me to explain to you what this code does?
https://github.com/openwrt/openwrt/blob/ac03e246351fbfa98e98f0319cce3a8663f836d3/include/image-commands.mk#L169-L176
I have not found the use of this variable anywhere else.

Reply:
"check-size" checks the size of the final image. For devices with small flash, typically 4M/8M, you will quickly manage to create images that are bigger than the available amount of firmware partition space to write them to. Since that might result in a broken or even unbootable device, the build routine checks the image size and will abort the build process if the "image" is too large when reaching the check-size check.